src/libraries/Microsoft.Extensions.Logging.Console/tests/Microsoft.Extensions.Logging.Console.Tests/ConsoleLoggerExtensionsTests.cs

```diff
@@ -565,5 +565,43 @@ public static TheoryData<string> FormatterNames
                 return data;
             }
         }
+
+        /// <summary>
+        /// Tests to ensure the suppression of IL3050 on ConsoleLoggerExtensions.AddConsole is valid.
+        /// </summary>
+        [Theory]
+        [InlineData(typeof(JsonConsoleFormatterOptions))]
+        [InlineData(typeof(ConsoleFormatterOptions))]
+        [InlineData(typeof(SimpleConsoleFormatterOptions))]
+        [InlineData(typeof(ConsoleLoggerOptions))]
+        public void EnsureFormatterOptions_OnlyHaveSimpleProperties(Type type)
+        {
+            VerifyHasOnlySimpleProperties(type);
+        }
+
+        private static void VerifyHasOnlySimpleProperties(Type type)
+        {
+            foreach (PropertyInfo prop in type.GetProperties())
+            {
+                if (type == typeof(JsonConsoleFormatterOptions) && prop.Name == "JsonWriterOptions")
+                {
+                    VerifyHasOnlySimpleProperties(prop.PropertyType);
+                    continue;
+                }
+
+                if (type == typeof(JsonWriterOptions) && prop.Name == "Encoder")
+                {
+                    // skip JsonWriterOptions.Encoder, since that can't be set through IConfiguration
+                    continue;
+                }
+
+                // verify only "simple" types are used in the Options classes, there can't be any generic collections
+                // or else NativeAOT would break
+                Assert.True(prop.PropertyType == typeof(string) ||
+                    prop.PropertyType == typeof(bool) ||
+                    prop.PropertyType == typeof(int) ||
```

Comment:
Are these the arguments used for MakeGeneric?

Instantiations over valuetypes need specialized code, so instantiating something over bool/int could fail (e.g. `typeof(List<>).MakeGenericType(typeof(bool))` will throw unless we already have the code for `List<bool>` guaranteed from elsewhere (what is guaranteeing it?).

Reply:
> Are these the arguments used for MakeGeneric?
No.

The way ConfigurationBinder works is that loops over the Type's properties and tries to "bind" the properties to an `IConfiguration` (i.e. `Dictionary<string, string>`).

The places that are AOT-unfriendly are handling the more complicated property types - like a `IList<T>` or `Dictionary<T, K>`. Those places need to use `MakeGenericType`. So if you had an object like:

```C#
        class OptionsWithLists
        {
            public IList<int> MyInts { get; set; }
        }
```

The ConfigurationBinder will see the `MyInts` property and try to create an `int[]` instance to fill in that property with all the values in the `IConfiguration`.

But here all the properties are simple, which is what this test is verifying. (Except the `Formatter` property on JsonWriterOptions, which isn't possible to bind using an `IConfiguration`.) Since they are all simple (string, bool, int), ConfigurationBinder just tries parsing the string to the destination type, which will work correctly.

When/if we have AOT tests like we do for TrimmingTests, this would be a good candidate to try to configure all the properties of these types and ensure it still works correctly.

Note - after looking into the existing IL2026 suppression in the code, I found out that we aren't actually preserving the properties correctly for trimming. I opened https://github.com/dotnet/runtime/issues/73822 for that issue.